File: kotlin_pro/src/cn/andios/kotlin/java2kotlin/JavaKotlinTest2.java
package cn.andios.kotlin.java2kotlin;

/**
 * @description:
 * @author:LSD
 * @when:2020/08/21/0:39
 */
public class JavaKotlinTest2 {
    // 我们无法通过new关键字来创建kotlin编译器自动生成的以Kt结尾的类的实例
    // 因为生成的字节码中没有不带参数的构造方法
//    JavaKotlinTest2Kt j = new JavaKotlinTest2Kt();

    MyClass myClass = new MyClass();

//    JavaKotlinTest2Kt.setStr("111")
//    System.out.println(JavaKotlinTest2Kt.test());
//
//    JavaKotlinTest2Kt.test();

}

File: kotlin_pro/src/cn/andios/kotlin/java2kotlin/JavaKotlinTest2.kt
package cn.andios.kotlin.java2kotlin

// 如果在kotlin文件中写了包级别的（不属于任何一个具体的类）属性或方法，那么这个方法或属性将成为
// 它所在的"文件名+Kt"类的静态成员变量和静态方法，调用时根据类名调用即可（方法直接调用，属性通过get/set调用）

// 反编译：public final class cn.andios.kotlin.java2kotlin.MyClass
class MyClass

// 反编译：public startic final void test();
fun test() {
    println("hello")
}

// 反编译：private static final java.lang.String str;
var str: String = "world"

